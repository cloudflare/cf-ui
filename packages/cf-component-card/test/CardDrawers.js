import React from 'react';
import renderer from 'react-test-renderer';
import {
  CardDrawers,
  CardToolbar,
  CardToolbarLink
} from '../../cf-component-card/src/index';

test('should render when given all the props', () => {
  const component = renderer.create(
    <CardDrawers
      onClick={() => {}}
      active="one"
      drawers={[
        { id: 'one', name: 'One', content: 'One Content' },
        { id: 'two', name: 'Two', content: 'Two Content' }
      ]}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render when not given props', () => {
  const component = renderer.create(
    <CardDrawers
      drawers={[
        { id: 'one', name: 'One', content: 'One Content' },
        { id: 'two', name: 'Two', content: 'Two Content' }
      ]}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
