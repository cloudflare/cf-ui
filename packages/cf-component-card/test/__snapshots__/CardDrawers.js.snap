// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render when given all the props 1`] = `
<div
  className="cf-card__section cf-card__section--default"
>
  <div
    className="cf-card__toolbar"
  >
    <div
      className="cf-card__toolbar_controls"
    />
    <div
      className="cf-card__toolbar_links"
      role="tablist"
    >
      <a
        className="cf-link cf-card__toolbar_link cf-card__toolbar_link--open"
        href="#!"
        id="card-0-drawer-one"
        onClick={[Function]}
        role="tab"
      >
        One
      </a>
      <a
        className="cf-link cf-card__toolbar_link"
        href="#!"
        id="card-0-drawer-two"
        onClick={[Function]}
        role="tab"
      >
        Two
      </a>
    </div>
  </div>
  <div
    className="cf-card__drawers_container cf-card__drawers_container--open"
  >
    <div
      aria-hidden="false"
      aria-labelledby="card-0-drawer-one"
      className="cf-card__drawer cf-card__drawer--active"
      role="tabpanel"
    >
      One Content
    </div>
    <div
      aria-hidden="true"
      aria-labelledby="card-0-drawer-two"
      className="cf-card__drawer"
      role="tabpanel"
    />
  </div>
</div>
`;

exports[`should render when not given props 1`] = `
<div
  className="cf-card__section cf-card__section--default"
>
  <div
    className="cf-card__toolbar"
  >
    <div
      className="cf-card__toolbar_controls"
    />
    <div
      className="cf-card__toolbar_links"
      role="tablist"
    >
      <a
        className="cf-link cf-card__toolbar_link"
        href="#!"
        id="card-1-drawer-one"
        onClick={[Function]}
        role="tab"
      >
        One
      </a>
      <a
        className="cf-link cf-card__toolbar_link"
        href="#!"
        id="card-1-drawer-two"
        onClick={[Function]}
        role="tab"
      >
        Two
      </a>
    </div>
  </div>
  <div
    className="cf-card__drawers_container"
  >
    <div
      aria-hidden="true"
      aria-labelledby="card-1-drawer-one"
      className="cf-card__drawer"
      role="tabpanel"
    />
    <div
      aria-hidden="true"
      aria-labelledby="card-1-drawer-two"
      className="cf-card__drawer"
      role="tabpanel"
    />
  </div>
</div>
`;
